### PR TITLE
Add StackedEnsemble models to h2o.

### PR DESCRIPTION
**Pull request by rpeck**

### CODE REVIEW

**Comment by ledell** [comment]
@h2o-ops please test!

**Comment by abal5** [comment]
@h2o-ops please test!

File: h2o-bindings/bin/gen_R.py [hunk 1]
```diff
@@ -430,7 +430,7 @@ def get_extra_params_for(algo):
     elif algo == "svd":
         return "training_frame, x, destination_key"
     elif algo == "stackedensemble":
-        return "x, y, training_frame, model_id"
```
Comment (mmalohlava) [inline h2o-bindings/bin/gen_R.py]: Is this change ok with respect to API compatibility.

File: h2o-core/src/main/java/hex/Model.java [hunk 2]
```diff
@@ -1072,6 +1072,11 @@ static protected void cleanup_adapt( Frame adaptFr, Frame fr ) {
     return names;
   }
 
+  /** Allow subclasses to define their own BigScore class. */
+  protected BigScore makeBigScore(String[][] domains, String[] names , Frame adaptFrm, boolean computeMetrics, Job j) {
+    return new BigScore(domains[0],names.length,adaptFrm.means(),_output.hasWeights() && adaptFrm.find(_output.weightsName()) >= 0,computeMetrics, true /*make preds*/, j).doAll(names.length, Vec.T_NUM, adaptFrm);
```
Comment (mmalohlava) [inline h2o-core/src/main/java/hex/Model.java]: THIS IS the big PROBLEM! 🔥 

the task is created, launched, but code below it relaunched it again!

File: h2o-core/src/main/java/hex/Model.java [hunk 3]
```diff
@@ -1101,9 +1108,12 @@ protected Frame predictScoreImpl(Frame fr, Frame adaptFrm, String destination_ke
...
     final boolean computeMetrics = (!isSupervised() || (adaptFrm.vec(_output.responseName()) != null && !adaptFrm.vec(_output.responseName()).isBad()));
     // Build up the names & domains.
-    String [] domain = !computeMetrics ? _output._domains[_output._domains.length-1] : adaptFrm.lastVec().domain();
+    String[] names = makeScoringNames();
+    String[][] domains = new String[names.length][];
+    domains[0] = names.length == 1 ? null : !computeMetrics ? _output._domains[_output._domains.length-1] : adaptFrm.lastVec().domain();
+
     // Score the dataset, building the class distribution & predictions
-    BigScore bs = new BigScore(domain,0,adaptFrm.means(),_output.hasWeights() && adaptFrm.find(_output.weightsName()) >= 0,computeMetrics, false /*no preds*/, null).doAll(adaptFrm);
+    BigScore bs = makeBigScore(domains, names, adaptFrm, computeMetrics, null).doAll(names.length, Vec.T_NUM, adaptFrm);
```
Comment (mmalohlava) [inline h2o-core/src/main/java/hex/Model.java]: This will leak.

**Pull request merged**